test/e2e/operator_test.go

```diff
@@ -70,172 +76,175 @@ var _ = Describe("Operator API", func() {
 	// 14. Ensure the reference to ns-a is eventually removed from o's status.components.refs field
 	// 15. Delete o
 	// 16. Ensure o is not re-created
-	It("should surface components in its status", func() {
-		o := &operatorsv1.Operator{}
-		o.SetName(genName("o-"))
-
-		Consistently(o).ShouldNot(ContainCopiedCSVReferences())
-
-		Eventually(func() error {
-			return client.Create(clientCtx, o)
-		}).Should(Succeed())
+	Context("when an Operator resource can select its components by label", func() {
+		var (
+			o *operatorsv1.Operator
+		)
+		BeforeEach(func() {
+			o = &operatorsv1.Operator{}
+			o.SetName(genName("o-"))
 
-		defer func() {
 			Eventually(func() error {
-				err := client.Delete(clientCtx, o)
-				if apierrors.IsNotFound(err) {
-					return nil
-				}
-
-				return err
+				return client.Create(clientCtx, o)
+			}).Should(Succeed())
+		})
+		AfterEach(func() {
+			Eventually(func() error {
+				return controllerclient.IgnoreNotFound(client.Delete(clientCtx, o))
 			}).Should(Succeed())
-		}()
+		})
 
-		By("eventually having a status that contains its component label selector")
-		w, err := operatorClient.Watch(clientCtx, listOpts)
-		Expect(err).ToNot(HaveOccurred())
-		defer w.Stop()
+		It("should not contain copied csv status references", func() {
+			Consistently(o).ShouldNot(ContainCopiedCSVReferences())
+		})
 
-		deadline, cancel := context.WithTimeout(clientCtx, 1*time.Minute)
-		defer cancel()
+		It("should surface referenced components in its status", func() {
+			By("eventually having a status that contains its component label selector")
+			w, err := operatorClient.Watch(clientCtx, listOpts)
+			Expect(err).To(BeNil())
+			defer w.Stop()
```

Comment:
```suggestion
			defer w.Stop()
			Expect(err).To(BeNil())
```
```suggestion
			Expect(err).To(BeNil())
			defer w.Stop()
```

Reply:
I updated the Context -> When given the comment structure was already there. Do you have any preference when to use a Context vs. When containers? I wasn't able to find any best practices around structuring ginkgo e2e tests so there might be something I'm missing.